Development progress - week 10

rayapetkova opened the pull request.

karastoyanov committed:
test file for PR

karastoyanov committed:
Major Fixes on Login and Register Menus
 - files relantions fixed and reworked
 - new login backed logic
 - database admin and client init functions from postgres_conn
   functionality is optimized. Clients are executed only once when
   starting the app

karastoyanov committed:
code rework

karastoyanov committed:
repo dir strucutre optimized

karastoyanov committed:
Postgres Dump Backup - 04.08

karastoyanov committed:
Dump Backup

karastoyanov committed:
Dump backup

rayapetkova committed:
When the 'favourites' button is clicked it shows the user's favourites items with default value (spin box)

- Added a new column 'quantity_wanted' in favourite_products' table which shows what quantity the user wants
- Implemented a QSpinBox in 'favourites.py' file which has a default value. The value comes from the inner join query (added a 'favourite_products.quantity_wanted' so we can get the wanted quantity)

rayapetkova committed:
Basic implementation of the 'basket' logic

karastoyanov committed:
typo fix

karastoyanov committed:
register form fixed

karastoyanov committed:
pop-out window on approved registration

karastoyanov committed:
register_user new logic

karastoyanov committed:
dynamic login fixed

karastoyanov committed:
fix dynamic login query for currently logged in user in main_menu

karastoyanov committed:
updated relations && fixed update user info functionality

karastoyanov committed:
products update

karastoyanov committed:
code cleaning and optimization

karastoyanov committed:
conflicts resolved partially

karastoyanov committed:
git revert

karastoyanov committed:
favorites and main_menu conflicts fixed

karastoyanov committed:
products conflicts fixed

karastoyanov committed:
Merge pull request #63 from Stake-And-Rope/files-assembly

Files assembly

karastoyanov committed:
final touches after conflicts resolving

karastoyanov committed:
basket sql query fix

rayapetkova committed:
Basket logic fixed

nikichhh committed:
Show password button logic and displayed added

The "Show password" button is for now only in the login.py and register.py files

nikichhh committed:
Log Out button added in the main_menu file

"Log Out" button logic and displayed added in the main_menu.py file

rayapetkova committed:
Update basket/favourites

karastoyanov committed:
minor changes

karastoyanov committed:
minor changes

karastoyanov committed:
payment options menu reworked

karastoyanov committed:
payment options update

karastoyanov committed:
DeleteClass fix

karastoyanov committed:
Create New Card labels created

karastoyanov committed:
payment options minor update

karastoyanov committed:
payment options delete card functionality fixed

karastoyanov committed:
add new payment option implemented

karastoyanov commented on pyqt5-gui/products.py:
```diff
@@ -96,15 +97,15 @@ def redirect_to_insert_to_postgre_func(prod_id, prod_name, spin_box):
         current_basket_button.setIconSize(QSize(30, 30))
         current_basket_button.setFont(QFont(fonts[0], 12))
+        current_basket_button.clicked.connect(redirect_to_insert_to_basket_func(product_id, product_name, current_spin_box, single_price))
 
```
@rayapetkova We will need to add addtional button -> "Delete From Basket"

karastoyanov commented on pyqt5-gui/favourites.py:
```diff
@@ -44,90 +35,90 @@ def redirect_to_delete_postgres_func(c_product_name):
+        current_basket_button.setFixedHeight(35)
+        current_basket_button.setIcon(QIcon(r'../img/shoppingcart.png'))
+        current_basket_button.setIconSize(QSize(30, 30))
+
```
@rayapetkova We will need to add additional button -> "Remove from favorites"

karastoyanov commented on pyqt5-gui/main_menu.py:
```diff
@@ -301,6 +321,15 @@ def open_categories():
+
+        """LOG OUT TO LOGIN SCREEN"""
+        def log_out():
+            # admin_cursor.close()
```
@karastoyanov Fix closing the server and database close connection after log-out

karastoyanov committed:
code review messages

nikichhh committed:
Added after clicking "Enter"/"Return" to do the same as clicked with mouse

I overrided the keyPressEvent function making the log_in_button to respond after clicking "Enter" or "Return" key (depends on the OS)

nikichhh committed:
Visual changes on some files

Small visual changes in the files: main_menu, subcategories, products, basket and favourites

rayapetkova committed:
Remove current spin box option in products menu (main)

karastoyanov committed:
hiding favorites and basket menu's fixed

karastoyanov committed:
remove favorites button - functinality not implemented

rayapetkova committed:
Update some functionalities

- The user can add a product from favourites to basket and from basket to favourites
- Added a remove button to the products in the basket
- Added a ToolTip to every button that refers to a product

karastoyanov committed:
Project Update:
 - 'Confirm Order' button added in basket menu
 - code optimizing and cleaning
 - test function for confirming the order -> confirm_order.py in
   db_handle dir

rayapetkova committed:
Confirm order (implemented button functionality)

karastoyanov committed:
conflicts fix
